main.go

```diff
@@ -115,13 +118,27 @@ func main() {
 	saInformer := informerFactory.Core().V1().ServiceAccounts()
 
 	*tokenExpiration = pkg.ValidateMinTokenExpiration(*tokenExpiration)
+
+	sess, err := session.NewSession()
+	if err != nil {
+		klog.Fatalf("Error creating session: %v", err.Error())
+	}
+
+	metadataClient := ec2metadata.New(sess)
+	identity, err := metadataClient.GetInstanceIdentityDocument()
+	if err != nil {
+		klog.Fatalf("Error getting instance identity document: %v", err.Error())
+	}
+
 	saCache := cache.New(
 		*audience,
 		*annotationPrefix,
 		*regionalSTS,
+		*composeRoleArn,
 		*tokenExpiration,
 		saInformer,
 		cmInformer,
+		identity,
```

Comment:
Passing the entire instance identity document to the cache seems excessive and leaky, can we instead create a struct which includes all the data required for this feature, called something like `ComposeRoleArn`, which contains `Enabled`, `Region`, `Partition` and `AccountId`?

Reply:
Ok I changed it slightly, I hope this is better now.